lib/Service/LangRopeService.php

```diff
@@ -46,7 +46,25 @@ private function getExApp() {
 		return $exApp;
 	}
 
-	public function indexSources(string $userId, array $sources): void {
+	public function deleteSources(string $userId, string ...$sourceNames): void {
+		if (count($sourceNames) === 0) {
+			return;
+	  }
+
+		$params = [
+			'userId' => $userId,
+			'sourceNames' => $sourceNames,
+		];
+
+		$exApp = $this->getExApp();
+		if ($exApp === null) {
+			return;
+		}
+		$this->requestToExApp($userId, $exApp, '/deleteSources', 'POST', $params);
+	}
+
+	// variable number of args to type check array $sources
+	public function indexSources(string $userId, Source ...$sources): void {
```

Comment:
No need for this, you can use docblock types. e.g.

```
/**
 * @param Source[] $sources
 * ...
 */

```

Reply:
That keeps it compatible with my other PR https://github.com/julien-nc/cwyd/pull/3